libs/order-list/src/lib/order-list-manager.tsx

```diff
@@ -0,0 +1,73 @@
+import { AsyncRenderer } from '@vegaprotocol/ui-toolkit';
+import { OrderList } from './order-list';
+import type { OrderFields } from './__generated__/OrderFields';
+import { useDataProvider } from '@vegaprotocol/react-helpers';
+import { ordersDataProvider, sortOrders } from './orders-data-provider';
+import { useCallback, useMemo, useRef } from 'react';
+import type { AgGridReact } from 'ag-grid-react';
+import type { OrderSub_orders } from './__generated__/OrderSub';
+import isEqual from 'lodash/isEqual';
+
+interface OrderListManagerProps {
+  partyId: string;
+}
+
+export const OrderListManager = ({ partyId }: OrderListManagerProps) => {
+  const gridRef = useRef<AgGridReact | null>(null);
+  const variables = useMemo(() => ({ partyId }), [partyId]);
+
+  // Apply updates to the table
+  const update = useCallback((delta: OrderSub_orders[]) => {
+    if (!gridRef.current) {
+      return false;
+    }
+
+    const update: OrderFields[] = [];
+    const add: OrderFields[] = [];
+
+    delta.forEach((d) => {
```

Comment:
We need to do same trick here as in provider update function
```
const sortedOrders = sortOrders(delta);
const incoming = uniqOrders(sortedOrders);
incoming .forEach((d) => {
```
or just `uniqOrders(sortOrders(delta)).forEach((d) => {`

Reply:
Added a prepareIncomingOrders function to be even a bit more DRY, its important the sort comes first so uniq grabs the correct order